docs/source/tools/speech_data_processor.rst

```diff
@@ -5,4 +5,134 @@ Speech Data Processor (SDP) is a toolkit to make it easy to:
+2. ``main.py`` script calls ``run_processors.py``, passing in your config.
+3. ``run_processors.py`` does the following:
+
+a. picks out the processors you wish to run (you can specify a subset of the processors in the config override, e.g. to avoid re-running time-consuming steps).
+b. if some of the processors have not had "output_manifest_file" or "input_manfiest_file" entries specified, SDP will automatically create temporary files for those.
+c. instantiate the processor classes using ``hydra.utils.instantiate``
+d. run the run-time processor tests by calling the ``processor.test()`` method.
+e. run the processing method (``processor.process()``) of each processor in order.
+  
+
+Layout of config YAML files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The YAML config file for processing a dataset must contain a key ``processors``, the value of which is a list. Each item in that list is expected to be a dictionary specifying a processor class, i.e. it must have a key ``_target_``, the value of which is a path to a "processor" class, and the remaining keys must be the kwargs necessary to instantiate that class with ``hydra.utils.instantiate()`` (c.f. https://hydra.cc/docs/advanced/instantiate_objects/overview/).
+
+SDP will run the processors specified in the ``processors`` list in the config file. It will also check for a ``processors_to_run`` key in the config file, which can be either the string "all", or any Python "slice" object.
+
+.. note:: 
+    SDP will run the processors in the order in which they are listed in the config YAML file. Make sure to list the processors in an order which makes sense, e.g. create an initial manifest first; make sure to run asr inference before doing any processing which looks at ``pred_text`` fields in the manifest.
+
+Processor classes
+-----------------
+
+
+**BaseProcessor**
+~~~~~~~~~~~~~~~~~~
+
+All processor classes inherit from the ``BaseProcessor`` class. This is a simple abstract class which has 2 empty methods: ``process()`` and ``test()``. 
+These serve to remind us that SDP essentially just runs ``test()`` on all processors, and then ``process()`` on all processors.
+
+``ASRInference`` is a child class of ``BaseProcessor``. It has a simple ``process()`` method which runs transcription on every utterance in the input_manifest.
+
+``WriteManifest`` is also a child class of ``BaseProcessor``. It has a simple ``process()`` method which saves a copy of the input manifest containing only the fields specified in ``fields_to_save``.
+
+**BaseParallelProcessor**
+~~~~~~~~~~~~~~~~~~~~~~~~~
+``BaseParallelProcessor`` inherits from the ``BaseProcessor`` class. Within its ``.process()`` method, it calls other methods and functions, which allow it to do more complex processing. 
+Most importantly, it calls its ``.process_dataset_entry(data_entry)`` method on every utterance in the manifest, and it does this in parallel, allowing for more efficient processing.
```

Comment:
. Before name is mistake or referring to method in some special way?

Reply:
Mistake, nothing special